drivers/spi/spi_nrfx_spi.c

```diff
@@ -271,23 +272,49 @@ static int init_spi(struct device *dev, const nrfx_spi_config_t *config)
 	return 0;
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int spi_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t new_state = *((const u32_t *)context);
+
+		if (new_state != current_state) {
+			switch (new_state) {
+			case DEVICE_PM_ACTIVE_STATE:
+				init_spi(dev);
+				configure(dev, get_dev_data(dev)->ctx.config);
+				break;
+
+			default:
```

Comment:
Shouldn't we return `-ENOTSUP` for `DEVICE_PM_FORCE_SUSPEND_STATE` (and any other unknown state that might be added in some future) instead of pretending that we handle it properly?